File: example/unity/Assets/Predict.cs
/* 
*   Function
*   Copyright © 2025 NatML Inc. All rights reserved.
*/

using UnityEngine;
using UnityEngine.UI;
using Function;

[Function.Function.Embed(PredictorTag)]
public class Predict : MonoBehaviour {

    public Text text;

    // Replace this with your predictor tag
    private const string PredictorTag = "@fxn/greeting";

    private async void Start () {
        var fxn = FunctionUnity.Create();
        var predictions = await fxn.Predictions.Create(
            tag: PredictorTag,
            inputs: new () { ["name"] = "Yusuf" }
        );
        text.text = predictions.results[0] as string;
    }
}

File: example/python/predict.py
# 
#   Muna
#   Copyright © 2025 NatML Inc. All Rights Reserved.
#

from muna import Muna

# Create a Muna client
muna = Muna(access_key="<ACCESS KEY>")

# Make a prediction
prediction = muna.predictions.create(
    tag="@fxn/greeting",
    inputs={ "name": "Yusuf" }
)

# Print the result
print(prediction.results[0])